src/d/actor/d_a_bst.cpp

```diff
@@ -4,213 +4,2399 @@
+        g_env_light.setLightTevColorType(model_2B8, &i_this->tevStr);
+
+        i_this->mpTexMtxAnimator->entry(model_2B8->getModelData());
+        i_this->mpTevRegAnimator->entry(model_2B8->getModelData());
+        i_this->field_0x02B8->entryDL();
+
+        i_this->mpTexMtxAnimator->remove(model_2B8->getModelData());
+        i_this->mpTevRegAnimator->remove(model_2B8->getModelData());
+
+        J3DModel* model_388 = i_this->field_0x0388;
+        for (u16 i = 0; i < model_2B8->getModelData()->getJointNum(); i++) {
+            model_388->setAnmMtx(i, model_2B8->getAnmMtx(i));
+        }
+        for (u16 i = 0; i < model_2B8->getModelData()->getWEvlpMtxNum(); i++) {
+            model_388->setWeightAnmMtx(i, model_2B8->getWeightAnmMtx(i));
+        }
+        mDoMtx_copy(i_this->field_0x02B8->getModel()->getAnmMtx(0), *calc_mtx);
+
+        shadow_pos.setall(0.0f);
+        MtxPosition(&temp, &shadow_pos);
+        shadow_pos.y += REG0_F(1) * 10.0f;
+
+        i_this->mShadowID = dComIfGd_setRealShadow(
+            i_this->mShadowID,
+            TRUE,
+            model_388,
+            &shadow_pos,
+            (REG0_F(2) * 10.0f) + 2000.0f,
+            0.0f,
+            &i_this->tevStr
+        );
+
+        J3DModel* model_2C8 = i_this->field_0x02C8;
+        i_this->mpBtkAnm->entry(model_2C8->getModelData());
+        i_this->mpBrkAnm->entry(model_2C8->getModelData());
+
+        g_env_light.settingTevStruct(TEV_TYPE_ACTOR, &i_this->home.pos, &i_this->mEnvLight);
+        g_env_light.setLightTevColorType(model_2C8, &i_this->mEnvLight);
+        mDoExt_modelUpdateDL(model_2C8);
+        if (i_this->field_0x02B4 == 0) {
+            switch (i_this->field_0x2E98) {
+                case 0:
+                    mDoGph_gInf_c::setBlureRate(i_this->field_0x2E98);
+                    mDoGph_gInf_c::onBlure();
+                    break;
+                case 1:
+                default:
+                    i_this->field_0x2E98 = 0;
+                    mDoGph_gInf_c::offBlure();
+                    break;
+            }
+
+            beam_draw(i_this);
+            // yup, this is a static var.
+            static cXyz center_pos(0.0f, 0.0f, 0.0f);
+
+            g_env_light.settingTevStruct(TEV_TYPE_BG0, &center_pos, &i_this->field_0x2F20);
+            g_env_light.setLightTevColorType(i_this->field_0x2FDC, &i_this->field_0x2F20);
+            i_this->field_0x2FE0->entry(i_this->field_0x2FDC->getModelData());
+
+            mDoExt_modelUpdateDL(i_this->field_0x2FDC);
+            g_env_light.setLightTevColorType(i_this->field_0x2FD0, &i_this->field_0x2F20);
+
+            i_this->field_0x2FD8->entry(i_this->field_0x2FD0->getModelData());
+            i_this->field_0x2FD4->entry(i_this->field_0x2FD0->getModelData());
+            mDoExt_modelUpdateDL(i_this->field_0x2FD0);
+        }
+        dSnap_RegistFig(DSNAP_TYPE_UNKC9, i_this, 1.0f, 1.0f, 1.0f);
```

Comment:
You could rename this in the header to `DSNAP_TYPE_BST` since it's no longer unknown